ivy/stateful/optimizers.py

```diff
@@ -359,9 +362,13 @@ def __init__(
         self._first_pass = True
         self._should_compile = False
 
+        Optimizer.__init__(
+            self, lr, inplace, stop_gradients, compile_on_next_step, device
+        )
+
```

Comment:
The True argument is necessary to be passed as True to always set the init_on_first_step value as True. This also ensures that compile_on_next_step is being passed in the correct position and so, it is not necessary to pass it as a keyword argument in this case. Since device follows fallback_to_non_compiled, it should be passed as a keyword argument (device=device).